torch/csrc/utils/tensor_new.cpp

```diff
@@ -553,21 +553,33 @@ Tensor legacy_sparse_tensor_generic_ctor_new(
   ParsedArgs<4> parsed_args;
   auto r = parser.parse(args, kwargs, parsed_args);
   if (r.idx == 0) {
+    TORCH_WARN_ONCE(
+        "torch.sparse.SparseTensor() is deprecated."
+        "  Please use torch.empty((0,), dtype=, layout=torch.sparse_coo).");
     auto deviceOptional = r.deviceOptional(0);
     check_legacy_ctor_device(dispatch_key, deviceOptional);
     return at::empty({0}, build_options(options, scalar_type, deviceOptional));
   } else if (r.idx == 1) {
+    TORCH_WARN_ONCE(
+        "torch.sparse.SparseTensor(cdata=x._cdata) is deprecated."
+        "  Please use torch.sparse_coo_tensor(x._indices(), x._values(), x.shape).");
     // NOLINTNEXTLINE(performance-no-int-to-ptr)
     auto cdata = reinterpret_cast<void*>(r.toInt64(0));
     return at::unsafeTensorFromTH(cdata, true);
   } else if (r.idx == 2) {
+    TORCH_WARN_ONCE(
+        "torch.sparse.SparseTensor(indices, values, *, device=) is deprecated."
+        "  Please use torch.sparse_coo_tensor(indices, values, dtype=, device=).");
     // Note: this signature doesn't have a dtype, even though it has a device;
     // it probably shouldn't have a device (we should infer it).
     auto deviceOptional = r.deviceOptional(2);
     check_legacy_ctor_device(dispatch_key, deviceOptional);
     at::OptionalDeviceGuard device_guard(deviceOptional);
     return at::sparse_coo_tensor(r.tensor(0), r.tensor(1));
   } else if (r.idx == 3) {
+    TORCH_WARN_ONCE(
```

Comment:
You could try turning these into errors to see if they trigger CI failures? That could be one way to find hidden invocations deep within the stack that'd result in annoying warnings to users.

Reply:
@cpuhrsch 
```
grep -r -I --exclude "*~" 'sparse[.]\w*Tensor' test/
```
reveals 52 usages of `torch.sparse.SparseTensor` while in `torch/` there are none.
So, turning warn-once to an error will guarantee CI failures.

There exist legacy tests that test `SparseTensor` constructors. Do we want to remove these tests?

Reply:
@pearu - yes if we're getting rid of these constructors let's also update those tests. Maybe we'll find a gap in functionality or uncover a bug.

Reply:
After replacing warn-once with an error, CI still became green. So, no uncovered bugs here.